packages/dd-trace/src/plugins/plugin.js

```diff
@@ -41,6 +41,10 @@ module.exports = class Plugin {
     storage.enterWith({ ...store, span })
   }
 
+  exit (ctx) {
+    storage.enterWith(ctx.parentStore)
```

Comment:
Should probably verify the store was actually enabled when parentStore was captured and, if not, return to disabled state. See: https://github.com/nodejs/node/blob/b058b20d41b718cc2c53c678663f1633194eacb5/lib/async_hooks.js#L364-L377